adapters/openweb/openweb.go

```diff
@@ -0,0 +1,186 @@
+package openweb
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type openwebImpExt struct {
+	OpenWeb openrtb_ext.ExtImpOpenWeb `json:"openweb"`
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	totalImps := len(request.Imp)
+	errors := make([]error, 0, totalImps)
+	sourceIdToImpIds := make(map[int][]int)
+	var sourceIds []int
+
+	for i := 0; i < totalImps; i++ {
+
+		sourceId, err := validateImpression(&request.Imp[i])
+
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		if _, ok := sourceIdToImpIds[sourceId]; !ok {
+			sourceIdToImpIds[sourceId] = make([]int, 0, totalImps-i)
+			sourceIds = append(sourceIds, sourceId)
+		}
+
+		sourceIdToImpIds[sourceId] = append(sourceIdToImpIds[sourceId], i)
+
+	}
+
+	totalReqs := len(sourceIdToImpIds)
+	if totalReqs == 0 {
+		return nil, errors
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	reqs := make([]*adapters.RequestData, 0, totalReqs)
+
+	imps := request.Imp
+	reqCopy := *request
+	reqCopy.Imp = make([]openrtb2.Imp, totalImps)
+	for _, sourceId := range sourceIds {
+		impIds := sourceIdToImpIds[sourceId]
+		reqCopy.Imp = reqCopy.Imp[:0]
+
+		for i := 0; i < len(impIds); i++ {
+			reqCopy.Imp = append(reqCopy.Imp, imps[impIds[i]])
```

Comment:
Could you please add a json test where there are two impressions with the same aid to ensure this loop works correctly for n>1?